content/source/docs/extend/writing-custom-providers.html.md

```diff
@@ -50,7 +50,7 @@ To start, create a file named `provider.go`. This is the root of the provider
 and should include the following boilerplate code:
 
 ```go
-package main
+package provider_name
```

Comment:
This feels to me like this will cause errors. We don't tell the user to create a `provider_name` package, nor is it imported anywhere. It looks like this guide assumes all provider code will be written in the `main` package, in which case the `package main` is the correct syntax here.

Reply:
Makes sense, I didn’t think about it in that way, I’m going to close the pr.